yare-core/src/main/java/com/sabre/oss/yare/core/internal/DefaultEngineController.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * MIT License
+ *
+ * Copyright 2018 Sabre GLBL Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.yare.core.internal;
+
+import com.sabre.oss.yare.core.EngineController;
+import com.sabre.oss.yare.core.EngineListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefaultEngineController implements EngineController {
+    private static final Logger log = LoggerFactory.getLogger(DefaultEngineController.class);
+    private List<EngineListener> listeners = new ArrayList<>();
+
+    @Override
+    public void stopProcessing() {
+        log.warn("Stopping processing...");
+        for (EngineListener listener : listeners) {
+            listener.onStopProcessing();
+        }
+    }
+
+    @Override
+    public void registerEngineListener(EngineListener listener) {
```

Comment:
I removed this method from interface. As I discussed with @MarcinLachowicz we decided to add factory and remove DefaultEngineController from builder, because clients shouldn't able to register stopProcessingListener and pass it to builder. See last commit.